go-controller/pkg/metrics/ovnkube_controller.go

```diff
@@ -619,6 +653,56 @@ func ipsecMetricHandler(table string, model model.Model) {
 	}
 }
 
+// MontiorIPsecTunnelsState will register a labeled metric for the ipsec tunnel status. It will keep
+// checking ipsec established status for each Geneve tunnel towards other remote node for every 5s
+// and update the metric with tunnel up or down status.
+func MontiorIPsecTunnelsState(stopChan <-chan struct{}, wg *sync.WaitGroup, nbClient libovsdbclient.Client,
+	ovsVsctl ovsClient, ipsec ipsecClient) {
+	prometheus.MustRegister(metricIPsecTunnelState)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-stopChan:
+				return
+			default:
+				nbGlobal := &nbdb.NBGlobal{}
+				nbGlobal, err := libovsdbops.GetNBGlobal(nbClient, nbGlobal)
+				if err != nil {
+					klog.Errorf("Error in getting the NBGlobal row from Northbound db, err: %v", err)
```

Comment:
do you really want to exit here? i thought youd only want to exit on shut down (stopCh closed)?  `MontiorIPsecTunnelsState` is exec only once.